lib/collection/src/collection/state_management.rs

```diff
@@ -77,6 +77,17 @@ impl Collection {
         {
             let mut config = self.collection_config.write().await;
 
+            if config.uuid != new_config.uuid {
```

Comment:
How about this edge case:

Imagine a user created a collection in the middle of a rolling upgrade to a Qdrant version including this change.

In that case I expect that only half of the peers have the UUID set.

If we'd then restore a consensus snapshot, it'll wipe the collection (None vs UUID).

Shall we be a bit more careful here and just compare Some's?

```suggestion
            if config.uuid.zip(new_config.uuid).map_or(false, |(a, b)| a != b) {
```

Reply:
I think it might be too much of an edge case, but good thinking. @generall wdyt?